torch/csrc/jit/script/compiler.cpp

```diff
@@ -1871,11 +1870,16 @@ struct to_ir {
           {rhs},
           {},
           self);
-
     } else {
-      throw ErrorReport(stmt.lhs())
-          << "left-hand side of augmented assignment to module "
-          << "parameters/buffers can only be tensor types";
+      const auto rhs = NamedValue(stmt.rhs().range(), emitExpr(stmt.rhs())).value(*method.graph());
```

Comment:
I dont think this correctly handles objects which define `__iadd__` (which is why tensors do an inplace update)